dask/array/core.py

```diff
@@ -1176,7 +1178,7 @@ def from_func(func, shape, dtype=None, name=None, args=(), kwargs={}):
     """
     if args or kwargs:
         func = partial(func, *args, **kwargs)
-    name = name or next(names)
+    name = name or 'from_func-' + tokenize((func, shape, dtype, args, kwargs))
```

Comment:
I've been using `sorted(kwargs.items(), key=lambda kv: kv[0])`

An alternative I was thinking about was to make tokenize variadic and bake these tricks into the function itself

```python
def _tokenize(_arg):
    if isinstance(_arg, Base):
        return _arg.name
    if isinstance(_arg, dict):
        return sorted(_arg.items(), key=lambda kv: kv[0])
    if isinstance(_arg, np.array):
        return ...
    ...

def tokenize(*args):
    return md5(str(tuple(map(_tokenize, args))).encode()).hex
```
 
This would probably be a nice refactor and avoid a lot of issues by others in the future.